src/model/model.ts

```diff
@@ -216,12 +216,32 @@ export class Model {
     return false;
   }
 
+  // addPolicies adds policy rules to the model.
+  public addPolicies(sec: string, ptype: string, rules: string[][]): boolean {
+    const ast = this.model.get(sec)?.get(ptype);
+    if (!ast) {
+      return false;
+    }
+
+    for (const rule of rules) {
+      if (this.hasPolicy(sec, ptype, rule)) {
+        return false;
+      }
+    }
+
+    for (const rule of rules) {
+      ast.policy.push(rule);
+    }
+
+    return true;
+  }
+
   // removePolicy removes a policy rule from the model.
   public removePolicy(sec: string, key: string, rule: string[]): boolean {
     if (this.hasPolicy(sec, key, rule)) {
       const ast = this.model.get(sec)?.get(key);
       if (!ast) {
-        return true;
+        return false;
       }
       ast.policy = _.filter(ast.policy, r => !util.arrayEquals(rule, r));
       return true;
```

Comment:
`return true` should mean a policy has been removed, but here we didn't check if a policy has been removed, please add a check on the length of `ast.policy`

Reply:
Here, we don't need the extra check, as we are sure that the policy rule is present in the policy and has been removed. This statement takes care of this:
``` if (this.hasPolicy(sec, key, rule)) ```

Reply:
ok seems like I ignored that line